### DISCUSSION TITLE
Matching tbprofiler mutations to WHO mutation catalogue

### DISCUSSION THREAD

**Discussion by taranewman** [jodyphelan/TBProfiler#266] [Q&A] [answered]
Hello,

I was wondering if you might know the best way to match the tbprofiler mutations to the mutation names in the WHO mutation catalogue excel sheet (https://www.who.int/publications/i/item/9789240028173). I'm interested in parsing some of the data from the catalogue to append to the tbprofiler mutation outputs. 

I see in https://github.com/jodyphelan/tbdb/blob/master/tbdb.csv that you have the "WHO Confidence" column along with the tbprofiler "Gene" and "Mutation" columns so I was wondering if you may already have a script that is able to parse the data from the WHO catalogue and if that is something you would be willing to share?

Thank you.

**Comment by jodyphelan** [comment, marked as answer]
Hi @taranewman 

The process of converting the WHO format to HGVS was done by @LennertVerboven for [this paper](https://pubmed.ncbi.nlm.nih.gov/36584023/). Scripts are available at https://github.com/LennertVerboven/WHO_catalogue_paper.

**Reply by taranewman** [reply]
Thank you, this is very much appreciated!

**Reply by taranewman** [reply]
Hi @jodyphelan 

I have a few mutations output by tbprofiler that I was unsure if they are referencing a specific WHO catalogue mutation.

For example, a tbprofiler mutation in my data is **gid c.115delC**.
In @LennertVerboven 's script, the HGVS output **gid c.115del** is connected to the  variant **gid_117_del_1_gcg_gc** from the WHO catalogue of mutations (for the STM antibiotic).

I was wondering if it would be correct to assume from this that **gid c.115delC** is the same mutation as **gid_117_del_1_gcg_gc** ?


I attached a csv file with the tbprofiler_output mutation in the first column and then the possibly related outputs from @LennertVerboven 's script in the remaining columns.
[tb_profiler_WHO_hvgs_output.csv](https://github.com/jodyphelan/TBProfiler/files/11127359/tb_profiler_WHO_hvgs_output.csv)


For cases where the hgvs notation only shows c.[base_position]del, does this mean that any base can come after the "del", ex. c.[base_position]del**G**, or c.[base_position]del**C** and it would be referring to the same WHO catalogue mutation?


Please let me know if there is any part of this that I can clarify or if there is somewhere else I should direct this question.

Thank you.

**Comment by LennertVerboven** [comment]
Hi @taranewman 


>For example, a tbprofiler mutation in my data is gid c.115delC.
>In @LennertVerboven 's script, the HGVS output gid c.115del is connected to the variant gid_117_del_1_gcg_gc from the WHO catalogue of mutations (for the STM antibiotic).
>I was wondering if it would be correct to assume from this that gid c.115delC is the same mutation as gid_117_del_1_gcg_gc ?

Yes, this is the same variant, the conversion is explained in the paper @jodyphelan linked earlier, but the gid gene (locus tag [Rv3919c](https://mycobrowser.epfl.ch/genes/Rv3919c)) is encoded on the complementary strand hence the decreasing number while the deletion lies further ahead.

>I attached a csv file with the tbprofiler_output mutation in the first column and then the possibly related outputs from @LennertVerboven 's script in the remaining columns.
>[tb_profiler_WHO_hvgs_output.csv](https://github.com/jodyphelan/TBProfiler/files/11127359/tb_profiler_WHO_hvgs_output.csv)

This list looks correct at first sight.

>For cases where the hgvs notation only shows c.[base_position]del, does this mean that any base can come after the "del", ex. c.[base_position]delG, or c.[base_position]delC and it would be referring to the same WHO catalogue mutation?

No, this is not the case, a deletion is solely determined by its position, the nucleotide is invariable at a position (i.e., the number indicates which nucleotide from the reference genome). The [HGVS](https://varnomen.hgvs.org/recommendations/DNA/variant/deletion/) notation for a deletion actually specifies not to list the nucleotide as that information is redundant. In our notation the nucleotide is listed, mostly due to the fact that this matched common practice within TB.

Hope that clears it up.

Best
Lennert

**Reply by taranewman** [reply]
Thank you very much Lennert!
